samtranslator/plugins/api/implicit_rest_api_plugin.py

```diff
@@ -61,6 +61,12 @@ def _process_api_events(  # noqa: too-many-arguments
 
             sam_expect(event_properties, event_id, "", is_sam_event=True).to_be_a_map("Properties should be a map.")
 
+            # if the API ID is not provided in the event source properties, this implies that SAM-T will
+            # construct an implicit API resource for this API event source, and we need to add tags to the
+            # implicit API resource if customers specify `PropagateTags` property
+            if self.API_ID_EVENT_PROPERTY not in event_properties:
+                self._add_tags_to_implicit_api_if_necessary(function, template)
+
             self._add_implicit_api_id_if_necessary(event_properties)  # type: ignore[no-untyped-call]
```

Comment:
Shouldn't these be together? i.e. check once and do all implicit API related stuff once.